src/lang.rs

```diff
@@ -1101,17 +1162,70 @@ fn var_list(input: Span) -> IResult<Span, Vec<String>> {
 fn parse(input: Span) -> IResult<Span, Positioned<InlineOperator>> {
     with_pos(
         tuple((
-            tag("parse").precedes(multispace1.precedes(req_quoted_string)),
+            tag("parse").precedes(multispace1),
+            opt(tag("regex").precedes(multispace1)),
+            with_pos(req_quoted_string),
             opt(multispace1.precedes(with_pos(pair(tag("from"), multispace1).precedes(expr)))),
-            tag("as").preceded_by(multispace1).precedes(var_list),
+            opt(with_pos(tag("as").preceded_by(multispace1).precedes(var_list))),
             opt(multispace1.precedes(with_pos(pair(tag("from"), multispace1).precedes(expr)))),
             opt(tag("nodrop").preceded_by(multispace1)).map(|nd| nd.is_some()),
         ))
-        .map(|(s, e1, fields, e2, no_drop)| InlineOperator::Parse {
-            pattern: Keyword::new_wildcard(s),
-            fields,
-            input_column: (e1, e2),
-            no_drop,
+        .map(|(_p, is_regex, s, e1, user_fields_opt, e2, no_drop)| {
+            let (pattern, fields) = if is_regex.is_some() {
+                let named_fields: Vec<String> = match regex::Regex::new(&s.value) {
+                    Err(regex_err) => {
+                        input
+                            .extra
+                            .report_error_for("invalid regular expression")
+                            .with_code_range(s.range, format!("{}", regex_err))
+                            .send_report();
+                        Vec::new()
+                    }
+                    Ok(re) => {
+                        let named_caps: Vec<String> = re
+                            .capture_names()
+                            .into_iter()
+                            .flatten()
+                            .map_into()
+                            .collect();
+                        let unnamed_capture_count = re.captures_len() - 1 - named_caps.len();
+
+                        if unnamed_capture_count > 0 {
+                            input
+                                .extra
+                                .report_error_for("regular expression for the parse operator cannot have unnamed captures")
+                                .with_code_range(s.range, format!(
+                                    "contains {} unnamed capture(s)", unnamed_capture_count))
+                                .with_resolution("name the capture groups by adding -- ?P<capname> -- right after the left parenthesis")
+                                .send_report();
+                            Vec::new()
+                        } else if let Some(user_fields) = user_fields_opt {
+                            input
+                                .extra
+                                .report_error_for("the parse regex operator does not accept user-defined field names")
+                                .with_code_range(user_fields.range, "remove this")
```

Comment:
I found this error message is a little confusing. What about something like:

"the parse regex operator does not support an `as` clause" or something like that.

It took me a bit to figure out what the user-defined field names actually meant.

The resolution could say something like: "remove the `as` clause. The named capture groups in the regular expression determine the output column"